src/lib.rs

```diff
@@ -381,25 +382,24 @@ where
     }
 }
 
-impl<T, P> AtomicPtr<T, domain::Global, P> {
+impl<T, F, P> AtomicPtr<T, F, P> {
     /// Loads the value from the stored pointer and guards it using the given hazard pointer.
     ///
     /// The guard ensures that the loaded `T` will remain valid for as long as you hold a reference
     /// to it.
     ///
-    /// Note that this method is _only_ available when using [`Domain<Global>`](domain::Global),
-    /// since it is a singleton, and thus is guaranteed to fulfill the safety requirement of
-    /// [`AtomicPtr::load`]. For all other domains, use [`AtomicPtr::load`].
-    pub fn safe_load<'hp>(
-        &'_ self,
-        hp: &'hp mut HazardPointer<'static, domain::Global>,
-    ) -> Option<&'hp T>
+    /// This method is a safe alternative to [`AtomicPtr::load`] available to
+    /// [`Domain::Singleton`](singleton families) since it is a singleton, and thus
+    /// is guaranteed to fulfill the safety requirement of [`AtomicPtr::load`]
```

Comment:
This doesn't read quite right to me. Or rather it's worded a little unclearly. It's safe for domains _with singleton families_, because the unsafety of the trait guarantees that there is only ever one instance of a `Domain` with a singleton family, and therefore loads and stores using such a family must be using the same (single) instance of that domain.